src/Extensions/MaintenanceProxyExtension.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace CWP\CWP\Extension;
+
+use BringYourOwnIdeas\Maintenance\Reports\SiteSummary;
+use SilverStripe\Core\Environment;
+use SilverStripe\Core\Extension;
+
+if (!class_exists(SiteSummary::class)) {
```

Comment:
FYI this does nothing as the class below is still compiled...

Reply:
Are you sure? This is a common pattern in SilverStripe

Reply:
Pretty sure... I just tested it with this gist: https://gist.github.com/ScopeyNZ/7ad38ff41f3bdc03dd2141474e0793df

Reply:
I find this information mildly perplexing.

Reply:
We discussed this yesterday - I'm pretty sure it's needed for when the ClassManifest gets regenerated, but only when extending a class that doesn't always exist. In this case it's not required since Extension is always there

Reply:
Well I'm learning a lot about the PHP compiler today. Logical expressions (non-declarative code) are ignored at compile time. The compiler considers all class definitions. If the extended class exists then the class is defined and the `if` statement does nothing. If the extended class does not exist but you have the `if` statement then the `if` statement is respected and you don't get an error unless you try to instantiate the class (that's doing the extending). If you don't have the `if` statement then PHP will fatal at compile time and let you know the extended class does not exist.

PHP...